packages/cli/src/index.js

```diff
@@ -83,14 +84,58 @@ const loadDotEnvDefaultsMiddleware = () => {
   })
 }
 
+const updateCheckerMiddleware = (argv) => {
+  const excludedCommands = ['update', 'upgrade', 'ts-to-js']
+  if (excludedCommands.includes(argv._[0])) {
+    return
+  }
+  if (updateCommand.shouldShowUpgradeAvailableMessage()) {
+    process.on('exit', () => {
+      updateCommand.showUpgradeAvailableMessage()
+    })
+  }
+  if (updateCommand.isUpdateCheckDue()) {
+    const backgroundUpdateBaseLogPath = path.join(
+      getPaths().base || '/tmp',
+      '.redwood'
+    )
+    const out = fs.openSync(
+      path.join(
+        backgroundUpdateBaseLogPath,
+        'background-update-checker.out.log'
+      ),
+      'a'
+    )
+    const err = fs.openSync(
+      path.join(
+        backgroundUpdateBaseLogPath,
+        'background-update-checker.err.log'
+      ),
+      'a'
+    )
+    const child = spawn('yarn', ['rw', 'update', '--silent'], {
+      detached: true,
+      stdio: ['ignore', out, err],
+      cwd: getPaths().base,
+      env: { ...process.env },
+      shell: process.platform === 'win32',
+    })
+    child.unref()
+  }
+}
+
 // eslint-disable-next-line no-unused-expressions
 yargs
   .scriptName('rw')
-  .middleware([
-    getCwdMiddleware,
-    loadDotEnvDefaultsMiddleware,
-    telemetryMiddleware,
-  ])
+  .middleware(
+    [
+      getCwdMiddleware,
+      loadDotEnvDefaultsMiddleware,
+      telemetryMiddleware,
+      !process.env.REDWOOD_BACKGROUND_UPDATES_DISABLED &&
+        updateCheckerMiddleware,
```

Comment:
Middleware makes sense, but we have to make sure this doesn't slow down the CLI too much. If so we maybe we only add it as middleware to certain commands, like dev. Still TBD, just something to keep in mind

Reply:
I had this concern in the forefront of my mind during the initial implementation.

I tested with the `yarn rw check` command. I timed the command 10 times without any of my update command additions present. I then repeated this timing test 10 times with my new additions and for both the case of when an update check was due and when one was not due. There was no notable difference in the execution time - hopefully as a result of spawning a child process to do all the work in the background. 

I haven't repeated this test for the latest version of the PR, so I'll need to redo this to confirm.